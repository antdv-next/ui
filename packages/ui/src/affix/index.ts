import type { App } from 'vue'
import Affix from './affix.vue'

export type { AffixProps } from './define'
Affix.install = (app: App) => {
  app.component('AAffix', Affix)
}
export {
  Affix,
}
