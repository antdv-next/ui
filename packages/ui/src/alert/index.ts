import type { App } from 'vue'
import Alert from './alert.vue'

export type { AlertProps } from './define'

Alert.install = (app: App) => {
  app.component('AAlert', Alert)
}

export {
  Alert,
}

export default Alert
