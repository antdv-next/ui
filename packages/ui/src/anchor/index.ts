import type { App } from 'vue'
import AnchorLink from './anchor-link.vue'
import Anchor from './anchor.vue'

export type { AnchorLinkBaseProps, AnchorProps } from './define'

Anchor.install = (app: App) => {
  app.component('AAnchor', Anchor)
  app.component('AAnchorLink', AnchorLink)
}

AnchorLink.install = (app: App) => {
  app.component('AAnchorLink', AnchorLink)
}

export {
  Anchor,
  AnchorLink,
}
export default Anchor
