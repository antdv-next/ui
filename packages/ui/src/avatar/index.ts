import type { App, Plugin } from 'vue'
import Avatar from './avatar.vue'
import Group from './group.vue'

export type { AvatarGroupProps, AvatarProps } from './define'

Avatar.install = function (app: App) {
  app.component(Avatar.name!, Avatar)
  app.component(Group.name!, Group)
  return app
}

Avatar.Group = Group

export { Avatar, Group as AvatarGroup }

export default Avatar as typeof Avatar
  & Plugin & {
    readonly Group: typeof Group
  }
