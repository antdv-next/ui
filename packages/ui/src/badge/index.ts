import type { App, Plugin } from 'vue'
import Badge from './badge.vue'
import Ribbon from './ribbon.vue'

export type { BadgeProps } from './define'

Badge.install = function (app: App) {
  app.component(Badge.name!, Badge)
  app.component(Ribbon.name!, Ribbon)
  return app
}

export { Badge, Ribbon as BadgeRibbon }

export default Badge as typeof Badge
  & Plugin & {
    readonly Ribbon: typeof Ribbon
  }
