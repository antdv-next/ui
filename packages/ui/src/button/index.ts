import type { App } from 'vue'
import Button from './button.vue'

export type { ButtonProps } from './define'
Button.install = (app: App) => {
  app.component('AButton', Button)
}

Button.__ANT_BUTTON = true

export {
  Button,
}
