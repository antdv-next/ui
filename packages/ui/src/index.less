@import './style/index.css';
@import './wave/style/index.less';
@import './affix/style/index.less';
@import './button/style/index.less';
@import './alert/style/index.less';
@import './divider/style/index.less';
@import './space/style/index.less';
@import './anchor/style/index.less';
@import './tooltip/style/index.less';
@import './popover/style/index.less';
@import './spin/style/index.less';
@import './qrcode/style/index.less';
@import './flex/style/index.less';
@import './popconfirm/style/index.less';
