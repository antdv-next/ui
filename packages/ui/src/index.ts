// 主题相关导出
export * from './theme/cssvar/genCssvar'
export * from './theme/cssvar/genStyle'
export * from './theme/cssvar/parseStyle'

// 主题接口类型
export type * from './theme/interface/alias'
export type * from './theme/interface/components'
export type * from './theme/interface/maps'
export type * from './theme/interface/presetColors'
export type * from './theme/interface/seeds'

// 主题算法和配置
export * from './theme/themes/compact'
export * from './theme/themes/dark'
export * from './theme/themes/default'
export * from './theme/themes/seed'
