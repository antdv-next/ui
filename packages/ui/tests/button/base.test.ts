import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { h } from 'vue'
import { Button } from '../../src'
import mountTest from '../shared/mountTest.ts'

describe('button', () => {
  mountTest(Button)
  mountTest(h(Button, {
    size: 'large',
  }))
  mountTest(h(Button, {
    size: 'small',
  }))

  it('renders correctly', () => {
    const wrapper = mount(h(Button, {}, {
      default: () => 'Follow',
    }))
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })
})
