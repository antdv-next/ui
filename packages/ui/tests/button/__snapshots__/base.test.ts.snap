// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button > renders correctly 1`] = `
"<button class="ant-btn ant-btn-color-default ant-btn-variant-outlined">
  <!-- If existIcon is true, render without animation -->
  <!-- Otherwise render with animation -->
  <transition-stub name="ant-btn-loading-icon-motion" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub><span>Follow</span>
</button>"
`;
